include/af/array.h

```diff
@@ -1573,7 +1574,9 @@ extern "C" {
     AFAPI af_err af_release_array(af_array arr);
 
     /**
-       Increments an \ref af_array reference count
+       Increments an \ref af_array reference count. The `out` pointer
+       need not be retained in situations where the reference count
+       is the only effect needed.
```

Comment:
I am confused here, what is out pointer retain ?

Reply:
The function has two side effects. One is to set the value of the `out` pointer, which is the function's first argument. The other is to increment the `af_array`'s reference count. In situations where the only desired effect is the reference count, the `out` pointer can be tossed without further side-effects occurring.

Reply:
I don't think this was strictly true, so I've removed this text.